test/extensions/one-tx-payment.js

```diff
@@ -111,9 +113,36 @@ contract("One transaction payments", accounts => {
 
       await fundColonyWithTokens(colony, token, INITIAL_FUNDING);
       await colony.moveFundsBetweenPots(1, 0, 0, d1.fundingPotId, d2.fundingPotId, WAD, token.address);
+      await oneTxExtension.makePaymentFundedFromDomain(1, 0, 1, 0, RECIPIENT, token.address, 10, 2, GLOBAL_SKILL_ID, { from: COLONY_ADMIN });
+    });
+
+    it("should allow a single-transaction to occur in a child domain, paid out from the root domain", async () => {
+      await colony.addDomain(1, 0, 1);
+
+      await fundColonyWithTokens(colony, token, INITIAL_FUNDING);
       await oneTxExtension.makePayment(1, 0, 1, 0, RECIPIENT, token.address, 10, 2, GLOBAL_SKILL_ID, { from: COLONY_ADMIN });
```

Comment:
I think this test is passing by mistake -- as the first child, domain 2 would have childSkillIndex 0, which means the bug I suspect exists is slipping through. Using domain 3 would catch this.